enhancements/helm3/console.md

```diff
@@ -0,0 +1,144 @@
+
+The default chart repository will be served from [Red Hat public GitHub repository](https://github.com/redhat-developer/redhat-helm-charts).
+
+New charts will be added or existing curated by submitting PRs against the above GitHub repo.
+
+### How would the UI discover the charts
+
+1. The UI would invoke `/api/helm/charts/index.yaml` endpoint to get [the list of all available charts](https://helm.sh/docs/topics/chart_repository/#the-index-file) so that the available charts can be rendered in the developer catalog. 
+
+2. The above endpoint would proxy requests to the configured charts repository
+
+In the initial phase, the charts repository would be a served of [redhat-helm-charts](https://redhat-developer.github.io/redhat-helm-charts) public GitHub repo.
+
+![Helm Charts Repo Service](../helm3/assets/charts-repo.png)
+
+### How would disconnected installs work 
+
+1. The user would need to 'clone' the content of the chart repository over the fence
+
+   * The public GitHub repository could be cloned into inside-the-network GitHub or Gitlab instance and configured to serve static content ( "Pages" ).
+   * The content of the chart repository could be crawled and served using a (containerized or external) HTTP server, e.g. nginx
+
+2. The URL serving the above static content would need to be passed to chart repository proxy running inside the cluster. 
+
+### Configuring Helm Chart Repository location
+
+Configuring Helm repository location could be modeled similar to [`OperatorSource`](https://github.com/operator-framework/operator-marketplace/blob/7d230952a1045624b7601b4d6e1d45b3def4cf76/deploy/crds/operators_v1_operatorsource_crd.yaml). 
+Due to future federated usecases ([ODC-2994](https://issues.redhat.com/browse/ODC-2994))), a cluster admin should be able to declare multiple chart repositories.
+
+A cluster-scoped, top-level `HelmRepository` resource)[https://github.com/openshift/api/tree/master/config/v1] should be added:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: HelmRepository
+metadata:
+  name: cluster
+spec:
+  url: http://my.chart-repo.org/stable
+
+  # optional and only needed for UI purposes
+  displayName: myChartRepo
+
+  # optional and only needed for UI purposes
+  description: my private chart repo
+
+  # set to true if need to be disabled temporarly
+  disabled: true
+```
+
+The console operator would watch for changes on it and reconfigure the chart repository proxy. The console backend already implements a few helm endpoints (including the chart proxy), but the future might require to extract them into a separate service (e.g. to make openshift more modular). Hence, it would be good to define the configuration as the top-level resource detaching API from the implementation. Cluster admin would easily discover the new resource and manage it either via CLI or through UI. 
```

Comment:
the modularity approach is fine with just a helm.openshift.io group similar to template.openshift.io.

Reply:
seems odd to me that the console operator would be responsible for configuring the chart repository proxy.

Is the chart repository proxy not going to have its own operator that manages its config?

Reply:
> the modularity approach is fine with just a helm.openshift.io group similar to template.openshift.io.

Not against, but do we then loose ability to configure/manage chart repos from UI?

> seems odd to me that the console operator would be responsible for configuring the chart repository proxy.

@bparees the statement was written after we already added the endpoints to the console backend.

> Is the chart repository proxy not going to have its own operator that manages its config?

If we ever decide to extract the introduced Helm endpoints from the console backend into a standalone service, we should also accompanied it with its own operator.